File: src/main/kotlin/com/wealthStack/bankstatement/ImportResult.kt
package com.wealthStack.bankstatement

import com.wealthStack.bankstatement.query.OperationDto

data class ImportResult(
    val message: String,
    val bankName: String,
    val fileName: String,
    val operationsImported: Int,
    val operations: List<OperationDto>
)

File: frontend/src/app/core/import-result.ts
import { Operation } from './operation';

/** Outcome of importing a statement, mirroring the backend `ImportResult`. */
export interface ImportResult {
  message: string;
  bankName: string;
  fileName: string | null;
  /** Newly inserted rows. */
  operationsImported: number;
  /** Existing rows updated because they matched an already-imported operation. */
  operationsOverwritten: number;
  operations: Operation[];
}
